src/schema/sources.ts

```diff
@@ -575,6 +586,25 @@ export const resolvers: IResolvers<any, Context> = {
         throw err;
       }
     },
+    removeSource: async (
+      _,
+      { sourceId }: { sourceId: string },
+      ctx,
+    ): Promise<GQLEmptyResponse> => {
+      const sourceMember = await ctx.con
+        .getRepository(SourceMember)
+        .findOneByOrFail({ sourceId, userId: ctx.userId });
+      if (sourceMember.role !== SourceMemberRoles.Owner) {
+        throw new ForbiddenError(
+          'Access denied! You do not have permission for this action!',
+        );
+      }
```

Comment:
I created a function called `ensureSourcePermissions`.
https://github.com/dailydotdev/daily-api/blob/a044d864444e2dde848a3e5a99b0c59541d7493e/src/schema/sources.ts#L319
I suggest extending its functionality to support the delete operation to make permissions check consolidated.

Reply:
@idoshamun Oh thanks for that, modified, let me know what you think.